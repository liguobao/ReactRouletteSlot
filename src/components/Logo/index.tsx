/*
 * @Author: wzi
 * @Date: 2018-03-05 10:34:41
 * @Last Modified by: wzi
 * @Last Modified time: 2018-09-17 18:04:11
 */
// import logo from '@assets/logo.png';
import { L } from '@config/LANG';
import React from 'react';
import styled from '@common/helper/styled-component';
import { shouldUpdate } from '@common/decorator/decorator';
// const LogoImg = styled.img.attrs({
//     src: logo,
// })`
//     width: 32px;
// `;
const WebName = styled.h1`
    color: #fff;
    font-family: Myriad Pro, Helvetica Neue, Arial, Helvetica, sans-serif;
    font-weight: 100;
`;

const Wrapper = styled.div`
    height: 50px;
    position: relative;
    line-height: 50px;
    padding-left: 24px;
    transition: all 0.3s;
    background: ${(props) => props.theme.primary};
    overflow: hidden;
    > * {
        vertical-align: middle;
        display: inline-block;
    }
`;
@shouldUpdate({})
export default class Logo extends React.Component<{}, {}> {
    render() {
        return (
            <Wrapper>
                {/* <LogoImg /> */}
                <WebName>{L('WebName')}</WebName>
            </Wrapper>
        );
    }
}
