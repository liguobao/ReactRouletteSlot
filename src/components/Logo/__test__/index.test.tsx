/*
 * @Author: wzi
 * @Date: 2018-01-31 16:40:02
 * @Last Modified by: wzi
 * @Last Modified time: 2018-04-27 21:52:21
 */
import React from 'react';
import Logo from '@components/Logo';
import { mount } from 'enzyme';
import 'jest-styled-components';
import toJson from 'enzyme-to-json';
describe('Logo', () => {
    let _props, _spies, _wrapper;
    beforeEach(() => {
        _spies = {};
        _props = {};
        _wrapper = mount(<Logo {..._props} />);
    });
    test('整体', () => {
        expect(toJson(_wrapper)).toMatchSnapshot();
    });
});
