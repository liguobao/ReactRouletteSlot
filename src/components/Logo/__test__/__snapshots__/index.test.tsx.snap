// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo 整体 1`] = `
.c1 {
  color: #fff;
  font-family: Myriad Pro,Helvetica Neue,Arial,Helvetica,sans-serif;
  font-weight: 100;
}

.c0 {
  height: 50px;
  position: relative;
  line-height: 50px;
  padding-left: 24px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  overflow: hidden;
}

.c0 > * {
  vertical-align: middle;
  display: inline-block;
}

<Logo>
  <styled.div>
    <div
      className="c0"
    >
      <styled.h1>
        <h1
          className="c1"
        >
          NB 彩票
        </h1>
      </styled.h1>
    </div>
  </styled.div>
</Logo>
`;
